includes/parserhooks/SetParserFunction.php

```diff
@@ -41,6 +46,15 @@ public function __construct( ParserData $parserData, MessageFormatter $messageFo
 	}
 
 	/**
+	 * @since  2.2
+	 *
+	 * @param Parser $parser
+	 */
+	public function setParser( $parser ) {
```

Comment:
Hm... you indeed don't run into a problem if the template param is not set. Still, I find this highly suspicious